app/src/main/java/io/pslab/communication/digitalChannel/DigitalChannel.java

```diff
@@ -67,18 +67,21 @@ public void loadData(LinkedHashMap<String, Integer> initialStates, double[] time
         double factor;
         switch (prescalar) {
             case 0:
-                factor = 64;
+                factor = 1.0/64.0;
                 break;
             case 1:
-                factor = 8;
+                factor = 1.0/8.0;
                 break;
             case 2:
+                factor = 1;
+                break;
+            case 3:
                 factor = 4;
                 break;
             default:
                 factor = 1;
         }
-        for (int i = 0; i < this.timestamps.length; i++) this.timestamps[i] /= factor;
+        for (int i = 0; i < this.timestamps.length; i++) this.timestamps[i] *= factor;
```

Comment:
Kinda same thing. Multiply and Divide is same as Divide and Multiply. Just FYI

Reply:
No but when prescalar is 3, we were before still divinding by 4, which according to Python library we are suppose to multiply . That's why I made this change.